auto-cite/auto-cite.py

```diff
@@ -60,36 +65,40 @@
 # go through sources
 for index, source in enumerate(sources):
     # show progress
-    log(f"Source {index + 1} of {len(sources)} - {source.get('id', '-')}", 2)
+    log(f"Source {index + 1} of {len(sources)} - {source.get('id', 'No ID')}", 2)
+
+    # new citation for source
+    new_citation = {}
 
     # find same source in existing citations
-    cached = find_match(source, citations)
+    cached = get_cached(source, citations)
 
     if cached:
         # use existing citation to save time
         log("Using existing citation", 3)
-        new_citations.append(cached)
+        new_citation = cached
 
-    else:
+    elif source.get("id", "").strip():
         # use Manubot to generate new citation
         log("Using Manubot to generate new citation", 3)
         try:
-            new_citations.append(cite_with_manubot(source))
+            new_citation = cite_with_manubot(source)
         except Exception as message:
             log(message, 3, "red")
             exit(1)
+    else:
+        # pass source through untouched
```

Comment:
You may want to elaborate on this comment some; it's not clear to me what the source is being passed through to. Is it just not adding extra citation data and leaving the citation as whatever was in `sources`?

Reply:
I think this is explained by the comments in the following line, and by other comments and docs. But we can discuss the wording of this particular comment later.